accname/name/comp_hidden_not_referenced.html

```diff
@@ -13,16 +14,119 @@
+>
+  <span aria-hidden="true">hidden,</span>
+  <span hidden>hidden from all users,</span>
+  <span>visible to all users</span>
+</button>
+
+<button
+  class="ex"
+  data-expectedlabel="hidden but referenced,"
+  data-testname="#2: button labelled by element that is hidden"
+  aria-labelledby="button-label-2"
+>
+  <span aria-hidden="true" id="button-label-2">hidden but referenced,</span>
+  <span hidden>hidden from all users,</span>
+  <span>visible to all users</span>
+</button>
+
+<button
+  class="ex"
+  data-expectedlabel="hidden from all users but referenced,"
+  data-testname="#3: button labelled by element that is hidden from all users"
+  aria-labelledby="button-label-3"
+>
+  <span aria-hidden="true">hidden,</span>
+  <span hidden id="button-label-3">hidden from all users but referenced,</span>
+  <span>visible to all users</span>
+</button>
+
+<button
+  class="ex"
+  data-expectedlabel="un-hidden and referenced indirectly,"
+  data-testname="#4: button labelled by hidden element with un-hidden child"
+  aria-labelledby="button-label-4"
+>
+  <span aria-hidden="true">hidden,</span>
+  <span aria-hidden="true" id="button-label-4">
+    <span aria-hidden="false" id="button-label-4-child">un-hidden and referenced indirectly,</span>
+  </span>
+  <span hidden>hidden from all users,</span>
+  <span>visible to all users</span>
+</button>
+
+<button
+  class="ex"
+  data-expectedlabel="un-hidden and referenced directly,"
+  data-testname="#5: button labelled directly by un-hidden child of hidden parent"
+  aria-labelledby="button-label-5-child"
+>
+  <span aria-hidden="true">hidden,</span>
+  <span aria-hidden="true" id="button-label-5">
+    <span aria-hidden="false" id="button-label-5-child">un-hidden and referenced directly,</span>
+  </span>
+  <span hidden>hidden from all users,</span>
+  <span>visible to all users</span>
+</button>
+
+<a
+  class="ex"
+  data-testname="#6: link labelled by multiple hidden and non-hidden elements"
+  data-expectedlabel="visible to all users, hidden but referenced, hidden from all users but referenced"
+  href="#"
+  aria-labelledby="link-label-1a link-label-1b link-label-1c"
+>
+  <span id="link-label-1a">
+    <span>visible to all users,</span>
+    <span aria-hidden="true">hidden,</span>
+  </span>
+  <span aria-hidden="true" id="link-label-1b">hidden but referenced,</span>
+  <span hidden id="link-label-1c">hidden from all users but referenced</span>
+</a>
+
+<h2
+  class="ex"
+  data-testname="#7: heading with name from content, containing content that is hidden with nested content that is un-hidden"
+  data-expectedlabel="visible to all users, un-hidden"
+>
+  visible to all users,
+  <span aria-hidden="true">
+    hidden,
+    <span aria-hidden="false">un-hidden</span>
+  </span>
+</h2>
+
+<h2
+  class="ex"
+  data-testname="#8: heading with name from content, containing content that is hidden from all users with nested content that is un-hidden for all users"
```

Comment:
hidden/un-hidden are ambiguous here too. I would recommend referencing `visibility:hidden` and `visibility:visible` directly in the testname for clarity...